app/models/ability.rb

```diff
@@ -0,0 +1,17 @@
+ class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user
+    can :read, :all
+    can [:update, :destroy, :create], [Comment]
```

Comment:
I have added check at the view page that if a author is able to modify/delete if he is either a author of that article or a admin.